client/dhcp_client_lwt.ml

```diff
@@ -0,0 +1,90 @@
+let src = Logs.Src.create "dhcp_client_lwt"
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Make(Time : Mirage_time_lwt.S) (Net : Mirage_net_lwt.S) = struct
+  open Lwt.Infix
+
+  type lease = Dhcp_wire.pkt
+
+  type t = lease Lwt_stream.t
+
+  let connect ?(with_xid)
+              ?(requests : Dhcp_wire.option_code list option) net =
```

Comment:
It is possible to build netif devices from various abstract stuff, for instance I have a small [small functor](https://github.com/mirage/mirage-net-flow) which tries to turn an arbitrary flow into a netif device. The connect function takes `mac` paramater, so all is fine. But sometimes, to get the `mac` parameter you need to do some other work (for instance, read in a database somewhere) so you might not now the mac address on startup. Right now, I have a rather complex startup sequence to make that work, so I was envisaging modifying my abstract netif to allow `set_mac` instead. But I was also wondering if there were another way :-)